compiler/rustc_lint/src/lints.rs

```diff
@@ -2077,7 +2077,19 @@ pub(crate) enum PathStatementDropSub {
 
 #[derive(LintDiagnostic)]
 #[diag(lint_path_statement_no_effect)]
-pub(crate) struct PathStatementNoEffect;
+pub(crate) struct PathStatementNoEffect {
+    #[subdiagnostic]
+    pub sub: PathStatementNoEffectSub,
+}
+
+#[derive(Subdiagnostic)]
+pub(crate) enum PathStatementNoEffectSub {
+    #[suggestion(lint_suggestion, code = "", applicability = "machine-applicable")]
+    Suggestion {
+        #[primary_span]
+        span: Span,
+    },
```

Comment:
A bit weird to have an enum with a single variant. Does this need to be separate from `PathStatementNoEffect`? `compiler/rustc_ast_lowering/src/errors.rs` looks like it has lots of errors with suggestions that are in a diagnostic, not a subdiagnostic.